server/utils/notifier.py

```diff
@@ -0,0 +1,177 @@
+"""
+Notifier.py
+
+Contains classes for various forms of notifying players, PC GMs, staff, and
+so on.  The purpose of these classes is to reduce code for sending messages
+to specific subsets of the game.  (e.g. - send only to staff in a given room)
+
+USAGE (example code):
+
+# This code will send "Hello, world!" then 'msg' to all player GMs
+# and staff GMs in the given room.
+gm_notifier = RoomNotifier(caller, room=caller.location, to_gm=True, to_staff=True)
+gm_notifier.notify("Hello, world!")
+gm_notifier.notify(msg)
+"""
+from typing import List
+
+from typeclasses.rooms import ArxRoom
+
+
+class NotifyError(Exception):
+    pass
+
+
+class Notifier:
+    """
+    Base class for sending notifications to the game.
+    This class is meant to be derived from and its ('private') code
+    utilized in derived classes (to lower code duplication).
+    """
+
+    def __init__(
+        self,
+        caller,
+        **to_flags,
+    ):
+        self.caller = caller
+        self.to_flags = to_flags
+
+        self.receiver_set = set()
+
+    def notify(self, msg: str, options: dict = None):
```

Comment:
Not that I care much, so this is just as an fyi, but having a type annotation with a default value of `None` will raise linting errors with more strict linters. The way around that is with the typing library, having a `Union[Dict, None]` as the type. But codacy didn't care so it's whatever